packages/unified-selection/src/unified-selection/iModel/IModelSelectionHandler.ts

```diff
@@ -0,0 +1,221 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/** @packageDocumentation
+ * @module UnifiedSelection
+ */
+
+import { from, Subject, takeUntil } from "rxjs";
+import { using } from "@itwin/core-bentley";
+import { CachingHiliteSetProvider } from "../CachingHiliteSetProvider";
+import { createHiliteSetProvider, HiliteSet, HiliteSetProvider } from "../HiliteSetProvider";
+import { IMetadataProvider } from "../queries/ECMetadata";
+import { IECSqlQueryExecutor } from "../queries/ECSqlCore";
+import { SelectableInstanceKey, Selectables } from "../Selectable";
+import { StorageSelectionChangeEventArgs, StorageSelectionChangeType } from "../SelectionChangeEvent";
+import { computeSelection, ElementSelectionScopeProps, SelectionScope } from "../SelectionScope";
+import { SelectionStorage } from "../SelectionStorage";
+import { IdArg, IModelConnection, SelectionSetEvent, SelectionSetEventType } from "./IModel";
+
+/** @internal */
+export interface IModelSelectionHandlerProps {
```

Comment:
Why not reuse `EnableUnifiedSelectionSyncWithIModelProps`?

Reply:
Reused.